uber/models.py

```diff
@@ -297,9 +297,11 @@ def _ints(self, name, val):
     @suffix_property
     def _label(self, name, val):
         try:
-            return self.get_field(name).type.choices[int(val)]
-        except KeyError:
-            return ''
+            val = int(val)
+        except ValueError:
```

Comment:
Can you add a log (probably WARN level, unless it's really something that
we don't mind happening and it can be DEBUG) whenever these automatic
silent fixes that "should never happen" are applied?

On Thu, Jun 16, 2016 at 11:45 AM, Victoria Earl <notifications@github.com>
wrote:

> In uber/models.py
> <https://github.com/magfest/ubersystem/pull/1868#discussion_r67370008>:
>
> > @@ -297,9 +297,11 @@ def _ints(self, name, val):
> >      @suffix_property
> >      def _label(self, name, val):
> >          try:
> > -            return self.get_field(name).type.choices[int(val)]
> > -        except KeyError:
> > -            return ''
> > +            val = int(val)
> > +        except ValueError:
>
> This is the new case we're handling, which is what happens when data gets
> munged. In fact, it's going to happen to Magstock once we switch over to
> the new campsite fields, because several attendees will have a string for
> site_number when we expect an int.
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/magfest/ubersystem/pull/1868/files/fe3e0585dee09940157098a445bd7ae6e94131c7#r67370008>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AAXeejWM9X8fALziheHyjSU9oNjmI1gCks5qMW-wgaJpZM4I3hco>
> .
>